gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopyableFile.java

```diff
@@ -441,7 +444,33 @@ public static Map<String, OwnerAndPermission> resolveReplicatedAncestorOwnerAndP
 
   private static List<AclEntry> getAclEntries(FileSystem srcFs, Path path) throws IOException {
     AclStatus aclStatus = srcFs.getAclStatus(path);
-    return aclStatus.getEntries();
+    return addOthersEntryTypeToAclEntriesIfMissing(aclStatus.getEntries());
+  }
+  /*
+   * When we get AclEntry from org.apache.hadoop.fs.permission.AclStatus, it's missing AclEntryType.OTHER.
+   * This causes AclTransformation validation to fail on the list of AclEntries. As a result, adding this helper
+   * method to provide DEFAULT scope in cases where AclEntryType.OTHER is absent
+   */
+  private static List<AclEntry> addOthersEntryTypeToAclEntriesIfMissing(List<AclEntry> aclEntries) {
+    // Check if "others" entry is missing
+    boolean othersAclEntryTypeMissing = true;
+
+    for (AclEntry aclEntry : aclEntries) {
+      if (aclEntry.getType() == AclEntryType.OTHER) {
+        othersAclEntryTypeMissing = false;
+        break;
+      }
+    }
+    // If "others" entry is missing, add it
+    if (othersAclEntryTypeMissing) {
```

Comment:
Can you help me understand why other can be missed in this case, and what is the best default value we should give? Also did we verify that user and group will never be missed in the ACL?

Reply:
with the recent change of using `modifyEntries` instead of `setAcl`... we would not require to check this since we will always assume we get the right list of acl entries